crates/core_arch/src/x86/sse.rs

```diff
@@ -1938,6 +1938,56 @@ extern "C" {
 /// exception _may_ be generated.
 ///
 /// [Intel's documentation](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_stream_ps)
+///
+/// # Safety of non-temporal stores
+///
+/// After using this intrinsic, but before any other access to the memory that this intrinsic
+/// mutates, a call to [`_mm_sfence`] must be performed.
```

Comment:
Does the `sfence` need to be executed from the same thread that initiated the non-temporal store? The proposed wording doesn't make that clear.

Reply:
Yes, it does. I will try to clarify the wording.